sdk/lib/mirrors/mirrors.dart

```diff
@@ -174,11 +174,17 @@ external ClassMirror reflectClass(Type key);
  * If [key] is not an instance of [Type], then this function throws an
  * [ArgumentError].
  *
+ * One can optionally pass a list of [typeArguments] for a generic class.
```

Comment:
"Optionally takes a list of [typeArguments] for generic classes. If the list is provided, then the [key] must be a generic class type, and the number of the provided type arguments must be equal to the number of type parameters declared by the class."

As a general advice, don't use future tense in dartdocs unless something happens after the function has returned.

Don't declare which error class is used when errors happen. Errors (contrary to exceptions) are bugs and should only happen during development. There should never be the need to catch them specifically. As such it is more flexible to give the implementations the room to throw different errors.